tests/compat.sh

```diff
@@ -205,7 +205,7 @@ filter()
 check_openssl_server_bug()
 {
     if test "X$VERIFY" = "XYES" && is_dtls "$MODE" && \
-        echo "$1" | grep "^TLS-PSK" >/dev/null;
+        test "$TYPE" = "PSK";
```

Comment:
Comparing the runs on Travis in https://app.travis-ci.com/github/Mbed-TLS/mbedtls/jobs/594300075 and https://app.travis-ci.com/github/Mbed-TLS/mbedtls/jobs/594178189, the logs look identical except that four test cases used to pass, but are now skipped:

Before:
```
m->O dtls12,yes TLS-DHE-PSK-WITH-ARIA-128-GCM-SHA256 ................... PASS
m->O dtls12,yes TLS-DHE-PSK-WITH-ARIA-256-GCM-SHA384 ................... PASS
m->O dtls12,yes TLS-DHE-PSK-WITH-CHACHA20-POLY1305-SHA256 .............. PASS
m->O dtls12,yes TLS-ECDHE-PSK-WITH-CHACHA20-POLY1305-SHA256 ............ PASS
```

After:
```
m->O dtls12,yes TLS_DHE_PSK_WITH_ARIA_128_GCM_SHA256 ................... SKIP
m->O dtls12,yes TLS_DHE_PSK_WITH_ARIA_256_GCM_SHA384 ................... SKIP
m->O dtls12,yes TLS_DHE_PSK_WITH_CHACHA20_POLY1305_SHA256 .............. SKIP
m->O dtls12,yes TLS_ECDHE_PSK_WITH_CHACHA20_POLY1305_SHA256 ............ SKIP
```

The problematic test cases are ones with (EC)DHE-PSK, so the old code didn't cause them to skip, but the new code does. According to the comment above the function, I think those test cases were supposed to be excluded, but they're passing. Presumably because the bug has been fixed in OpenSSL.

As this is partly a preexisting problem, I'm ok with keeping this discrepancy, and resolving it in a follow-up task.

Reply:
Right, this happens as we only skipped string with format of `*TLS-PSK*.` But `TLS-XXX-PSK` doesn't meet this requirement. So the check logic is different from old version. Should we restore the changes to keep test coverage identical as before then resolve this problem in the follow-up task? Or should we go ahead with those changes?

Reply:
Actually, thinking about it, PSK cipher suites (including (EC)DHE-PSK ones) don't care about certificates at all, so the test cases with `VERIFY=YES` duplicate the test cases with `VERIFY=NO`. And so there's no loss of test coverage here. We're just effectively running those four test cases once instead of twice.